source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -531,13 +459,32 @@ FilterConfigPerRoute::FilterConfigPerRoute(
       grpc_service_(initGrpcService(config)),
       grpc_initial_metadata_(config.overrides().grpc_initial_metadata().begin(),
                              config.overrides().grpc_initial_metadata().end()),
-      untyped_forwarding_namespaces_(initUntypedForwardingNamespaces(config)),
-      typed_forwarding_namespaces_(initTypedForwardingNamespaces(config)),
-      untyped_receiving_namespaces_(initUntypedReceivingNamespaces(config)),
+      untyped_forwarding_namespaces_(
```

Comment:
Can we simplify this code block a little bit, like put these in a helper function, and share some common code, and call a function for each of these 5 variables?

Reply:
the nullptr denotes the previous "null_opt". 
when we call the helper function, we need to pass in a boolean decided by "config.overrides().metadata_options().has_forwarding_namespaces()", and then the actual config.overrides().metadata_options().forwarding_namespaces().untyped(), which will be equally messy.

There is a somewhat space to tidy up a little bit, also asked by Adi: we can wrap the typed and the untyped using one data structure, but allow me to do that in next PR.